feat: adds $compute

johnsonav1992 opened the pull request:
This was a fun one! Here is an implementation of the `$compute` operator added to the library. I wrote a good number of tests and made sure to find as much information about the spec from both OData docs and other articles to make sure these use-cases are valid. I also spun up a separate OData service and ran a bunch of these test queries and some others to be sure they worked in practice. But feel free to pick it apart!

As a sidenote - I noticed the code in this package is not getting minified on build. Adding that in before releasing 3.0 might be super helpful to reduce the bundle size (though it isn't huge to begin with) and make it even faster and more efficient. Just a thought!

rosostolato commented:
Great! Good catch with the build minification, I will review it before pushing version 3, thank you!

rosostolato commented on README.md:
````diff
@@ -75,6 +75,7 @@ const complexQuery = odataQuery<User>()
 ```
 
 - [Filtering with `filter`](#filtering-with-filter)
+- [Computing with `compute`](#computing-with-compute)
````
Could you please move compute to the end after expand?

rosostolato commented:
@johnsonav1992 I posted here a complete review from claude, do you think there's anything to update based on this?

rosostolato commented:
> @rosostolato let me find a bit of time here soon this morning to look through it and see!

Sounds good!

About the proxy, I'm not sure, but I remember that I used proxy before and can't remember why I went back to literal props... I can't remember if something had broken, maybe it was a minification issue... I really can't remember

johnsonav1992 commented:
> > @rosostolato let me find a bit of time here soon this morning to look through it and see!
> 
> Sounds good!
> 
> About the proxy, I'm not sure, but I remember that I used proxy before and can't remember why I went back to literal props... I can't remember if something had broken, maybe it was a minification issue... I really can't remember
> 
> -- edit --
> 
> Actually we do have proxies for expand, filter, orderBy and select. But they are getters for the user's input model props.
> 
> Anyway, if we can confirm that it will work after minify the code, I think it's good to go as it is.

Yep! And that's why I stuck with a Proxy as well. I'll take a look and see what it means...I believe it's just referring to what method has the proxy in relation to the other ones so I can try to line it up better for consistency

johnsonav1992 commented:
@rosostolato Alright! I made some changes:

1. Updated the function names to match better with the other operator implementations
2. Handled more edge cases both in the code and in the tests for some mathematical and boolean operators
3. Made sure all the type concerns it had were either resolved or a non-issue
4. Updated the readme

Let me know what else we need!

rosostolato commented:
Looks good, thank you @johnsonav1992
